docs/docs/tutorials/adding_annotations.md

```diff
@@ -0,0 +1,184 @@
+
+# Adding the grpc-gateway annotations to an existing protobuf file
+
+Now that we've got a working Go gRPC server, we need to add the grpc-gateway annotations.
+
+The annotations define how gRPC services map to the JSON request and response. When using protocol buffers, each RPC must define the HTTP method and path using the `google.api.http` annotation.
+
+So you will need to add `import "google/api/http.proto";` to the gRPC proto file.
+
+```proto
+syntax = "proto3";
+
+package helloworld;
+
+import "google/api/annotations.proto";
+
+// Here is the overall greeting service definition where we define all our endpoints
+service Greeter {
+  // Sends a greeting
+  rpc SayHello (HelloRequest) returns (HelloReply) {
+    option (google.api.http) = {
+      post: "/v1/example/echo"
+      body: "*"
+    };
+  }
+}
+
+// The request message containing the user's name
+message HelloRequest {
+  string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+  string message = 1;
+}
+```
+
+HttpRule is typically specified as an `google.api.http` annotation on the gRPC method. Each mapping specifies a URL path template and an HTTP method.
+
+Also, See [a_bit_of_everything.proto](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/internal/proto/examplepb/a_bit_of_everything.proto) for examples of more annotations you can add to customize gateway behavior and generated OpenAPI output.
+
+## Generating the grpc-gateway stubs
+
+Now that we've got the grpc-gateway annotations added to the proto file, we need to use the grpc-gateway generator to generate the stubs.
+
+Before we can do that, we need to copy some dependencies into our protofile structure. Copy the `third_party/googleapis` folder from the grpc-gateway repository to your local protofile structure. It should look like this afterwards:
+
+```
+proto
+├── google
+│   └── api
+│       ├── annotations.proto
+│       └── http.proto
+└── helloworld
+    └── hello_world.proto
+```
+
+### Using buf
+
+We'll need to add the grpc-gateway generator to the generation configuration:
+
+```yml
+version: v1beta1
+plugins:
+  - name: go
+    out: proto
+    opt: paths=source_relative
+  - name: go-grpc
+    out: proto
+    opt: paths=source_relative,require_unimplemented_servers=false
+  - name: grpc-gateway
+    out: proto
+    opt: paths=source_relative
+```
+
+And that's it! Now if you run:
+
+```sh
+$ buf generate
+```
+
+It should produce a `*.gw.pb.go` file.
+
+### Using protoc
+
+Now we need to add the grpc-gateway generator to the protoc invocation:
+
+```sh
+$ protoc -I ./proto \
+  --go_out ./proto --go_opt paths=source_relative \
+  --go-grpc_out ./proto --go-grpc_opt paths=source_relative \
+  --grpc-gateway_out ./proto --grpc-gateway_opt paths=source_relative \
+  ./proto/helloworld/hello_world.proto
+```
+
+This should generate a `*.gw.pb.go` file.
+
+Usage examples can be found on this [Usage](https://github.com/grpc-ecosystem/grpc-gateway#usage)
```

Comment:
Lets remove this line, it seems a bit disconnected from the rest of the content.
```suggestion
```